bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java

```diff
@@ -181,50 +181,59 @@ static public boolean sendTelegramPhoto(@ParamDoc(name = "group") String group,
 
-        GetMethod getMethod = new GetMethod(photoURL);
-        getMethod.getParams().setSoTimeout(timeoutMillis);
-        getMethod.getParams().setParameter(HttpMethodParams.RETRY_HANDLER,
-                new DefaultHttpMethodRetryHandler(retries, false));
-        try {
-            int statusCode = getClient.executeMethod(getMethod);
-            if (statusCode != HttpStatus.SC_OK) {
-                logger.warn("Failed to retrieve an image. Received status: {}", getMethod.getStatusLine());
-                return false;
+            if (username != null && password != null) {
+                getClient.getParams().setAuthenticationPreemptive(true);
+                Credentials defaultcreds = new UsernamePasswordCredentials(username, password);
+                getClient.getState().setCredentials(AuthScope.ANY, defaultcreds);
             }
 
-            // if the content-length is 0 (which shouldn't happen),
-            // flag an appropriate error
-            if (getMethod.getResponseContentLength() == 0) {
-                logger.warn("Failed to retrieve an image. Fetched URL returned no data.");
+            GetMethod getMethod = new GetMethod(photoURL);
+            getMethod.getParams().setSoTimeout(timeoutMillis);
+            getMethod.getParams().setParameter(HttpMethodParams.RETRY_HANDLER,
+                    new DefaultHttpMethodRetryHandler(retries, false));
+            try {
+                int statusCode = getClient.executeMethod(getMethod);
+                if (statusCode != HttpStatus.SC_OK) {
+                    logger.warn("Failed to retrieve an image. Received status: {}", getMethod.getStatusLine());
+                    return false;
+                }
+
+                // if the content-length is 0 (which shouldn't happen),
+                // flag an appropriate error
+                if (getMethod.getResponseContentLength() == 0) {
+                    logger.warn("Failed to retrieve an image. Fetched URL returned no data.");
+                    return false;
+                }
+
+                image = getMethod.getResponseBody();
+            } catch (HttpException e) {
+                logger.warn("HTTP protocol violation: {}", e);
                 return false;
+            } catch (IOException e) {
+                logger.warn("Transport error: {}", e);
+                return false;
+            } finally {
+                getMethod.releaseConnection();
             }
-
-            imageFromURL = getMethod.getResponseBody();
-        } catch (HttpException e) {
-            logger.warn("HTTP protocol violation: {}", e);
-            return false;
-        } catch (IOException e) {
-            logger.warn("Transport error: {}", e);
-            return false;
-        } finally {
-            getMethod.releaseConnection();
+        } else {
+            // Load image from provided base64 image
+            logger.warn("Photo base64 provided, converting to binary.");
```

Comment:
Please consider, "Photo base64 provided; converting to binary."  <- I changed the comma to a semicolon.